### PR TITLE
quarantine_report.php and lists.php localisation

### PR DESCRIPTION
**Pull request by stefaweb**
Please test quarantine_report.php

---
This change is [Reviewable](https://reviewable.io/reviews/mailwatch/1.2.0/340)

### CODE REVIEW

File: tools/Cron_jobs/quarantine_report.php [hunk 1]
```diff
@@ -51,7 +51,7 @@
 $required_constant_missing_count = 0;
 foreach ($required_constant as $constant) {
     if (!defined($constant)) {
-        echo "The variable $constant is empty, please set a value in conf.php.\n";
+        echo __('message61') . "\n";
```
Comment (endelwar) [inline tools/Cron_jobs/quarantine_report.php]: `$constant` is a php variable, this need to be wrapper in sprintf

Comment (stefaweb) [inline tools/Cron_jobs/quarantine_report.php]: This?
sprintf(__('message61', true), $constant));

Comment (endelwar) [inline tools/Cron_jobs/quarantine_report.php]: line 54 should be `echo sprintf(__('message61'), $constant) . "\n";` and `message61` in translations array should be `The variable %s is empty, please set a value in conf.php.`.

`%s` is replaced by the content of `$constant`

**Comment by asuweb** [comment]
*[tools/Cron_jobs/quarantine_report.php, line 437 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6OFuHAfoV-0cbFQM:-K_W6OFuHAfoV-0cbFQN:b-kbafnm) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L437)):*
> __('view61') is the wrong translation, this should be changed to __('release61') and translate to "Release"  Need to add 'release61' to language files

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by asuweb** [comment]
*[tools/Cron_jobs/quarantine_report.php, line 424 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W5rdm5hkpGX7bOvs5:-K_W5rdm5hkpGX7bOvs6:b-in4gq9) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L424)):*
> ```PHP
>     foreach ($quarantined as $qitem) {
>         //Check if auto-release is enabled
>         $links = '<a href="' . QUARANTINE_REPORT_HOSTURL . '/viewmail.php?id=' . $qitem['id'] . '">'.__('arview01').'</a>';
> ```

__('arview01') doesn't exist in language files - this should be changed to __('view61') which translates as "View" in the language files

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by asuweb** [comment]
*[mailscanner/languages/en.php, line 734 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6pz-Na46XO2uVfvh:-K_W6pz-Na46XO2uVfvi:b-btqwpn) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/mailscanner/languages/en.php#L734)):*
> ```PHP
>     'text611' => 'Quarantine Report for %s',
>     'text612' => 'In the last %s day(s) you have received %s e-mails that have been quarantined and are listed below. All messages in the quarantine are automatically deleted %s days after the date that they were received.',
> ```

add entry for 'release61' => 'Release'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by stefaweb** [comment]
Not sure if I should translate this:

    /*
    ** Text Template
    */

    $text = 'Quarantine Report for %s

In the last %s day(s) you have received %s e-mails that have been quarantined and are listed below. All messages in the quarantine are automatically deleted %s days after the date that they were received.

%s';

    $text_content = 'Received: %s
From: %s
Subject: %s
Reason: %s
Action:
%s

';

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[mailscanner/languages/en.php, line 734 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6pz-Na46XO2uVfvh:-K_YQyZrAA7Gznbknzl1:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/mailscanner/languages/en.php#L734)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

add entry for 'release61' => 'Release'
</blockquote></details>

Done

---

*[tools/Cron_jobs/quarantine_report.php, line 54 at r1](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_UOCXDU58J9UZ1pZV1-r1-54:-K_YR468tTdzrEj0L_QC:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/b41dc92bebb03d2a696fd9b9dc5bf7dfb84d7527/tools/Cron_jobs/quarantine_report.php#L54)):*
<details><summary><i>Previously, endelwar (Manuel Dalla Lana) wrote…</i></summary><blockquote>

line 54 should be `echo sprintf(__('message61'), $constant) . "\n";` and `message61` in translations array should be `The variable %s is empty, please set a value in conf.php.`.

`%s` is replaced by the content of `$constant`
</blockquote></details>

Done

---

*[tools/Cron_jobs/quarantine_report.php, line 424 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W5rdm5hkpGX7bOvs5:-K_YR5pabxl9Nm0PQ6Ow:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L424)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

__('arview01') doesn't exist in language files - this should be changed to __('view61') which translates as "View" in the language files
</blockquote></details>

Done

---

*[tools/Cron_jobs/quarantine_report.php, line 437 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6OFuHAfoV-0cbFQM:-K_YR6kFurnK92Fj5yzC:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L437)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

> __('view61') is the wrong translation, this should be changed to __('release61') and translate to "Release"  Need to add 'release61' to language files
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[mailscanner/languages/en.php, line 734 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6pz-Na46XO2uVfvh:-K_YT14MvU7qIThwQqvJ:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/mailscanner/languages/en.php#L734)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---

*[tools/Cron_jobs/quarantine_report.php, line 54 at r1](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_UOCXDU58J9UZ1pZV1-r1-54:-K_YT3XozCuRkp3K93KD:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/b41dc92bebb03d2a696fd9b9dc5bf7dfb84d7527/tools/Cron_jobs/quarantine_report.php#L54)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---

*[tools/Cron_jobs/quarantine_report.php, line 424 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W5rdm5hkpGX7bOvs5:-K_YT4Idu4fv5kHJkRfI:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L424)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---

*[tools/Cron_jobs/quarantine_report.php, line 437 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6OFuHAfoV-0cbFQM:-K_YT4t5r8UmcMX3uv6b:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L437)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[mailscanner/languages/en.php, line 734 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6pz-Na46XO2uVfvh:-K_YSNYZwG2t7Iwd0zf_:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/mailscanner/languages/en.php#L734)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by asuweb** [comment]
@stefaweb - Is there a planned release date for the develop branch to move to master?  

I've a few features I'll be working on in the next month or so, and as there is a large code base change (which is excellent by the way, fantastic work from the mailwatch team), I'd rather work on the new code base rather than have to re-write the modifications twice, assuming it's release is in the next month?

Thanks

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[mailscanner/languages/en.php, line 734 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6pz-Na46XO2uVfvh:-K_YU0qptazWJCYvUdqO:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/mailscanner/languages/en.php#L734)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---

*[tools/Cron_jobs/quarantine_report.php, line 437 at r2](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_W6OFuHAfoV-0cbFQM:-K_YU7XCH9PEI07Zh7ta:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/0b7a2a15f7d0634324e9c8ca8a32eb13b8e07125/tools/Cron_jobs/quarantine_report.php#L437)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by stefaweb** [comment]
How to "resolve discussions" to have the PR in green?

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/Cron_jobs/quarantine_report.php, line 54 at r1](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_UOCXDU58J9UZ1pZV1-r1-54:-K_YV4nJz3JsF3a1gowF:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/b41dc92bebb03d2a696fd9b9dc5bf7dfb84d7527/tools/Cron_jobs/quarantine_report.php#L54)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by stefaweb** [comment]
Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/Cron_jobs/quarantine_report.php, line 54 at r1](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-K_UOCXDU58J9UZ1pZV1-r1-54:-K_YVHgygqtLmkqXCCmR:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/b41dc92bebb03d2a696fd9b9dc5bf7dfb84d7527/tools/Cron_jobs/quarantine_report.php#L54)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by asuweb** [comment]
The unresolved discussion needs to be resolved - on Reviewable, click "To Review" (top right), and then click "Acknowledge" under the discussions at that marker.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-:-K_YViIBV3_E2SjduEwU:bn2301)*

**Comment by stefaweb** [comment]
endelwar need to do it.

**Comment by stefaweb** [comment]
"Is there a planned release date for the develop branch to move to master?"
Don't know. You should ask endelwar.
It will may be better to "stabilise" 1.2 and publish it before adding new features.

**Comment by asuweb** [comment]
Thanks, I'll do that :)

**Comment by stefaweb** [comment]
@endelwar
How to find help to localize language files?

**Comment by endelwar** [comment]
Reviewed 1 of 9 files at r1, 1 of 8 files at r2, 8 of 8 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340)*

**Comment by endelwar** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/340#-:-K_Ywq09vjaoC0dMS7zV:bnfp4nl)*

**Pull request merged**